DOC: prepare changelog for 0.14.0 release

dnicolodi opened the pull request:
I've started to put together the changelog entry for the next release.

rgommers commented on CHANGELOG.rst:
```diff
@@ -11,6 +11,28 @@
+
+- Obey the ``--skip-subprojects`` option passed to the ``meson
+  install`` command.
+- Implement support for building wheels targerting the Python limited
```
typo in "targeting"

rgommers commented on CHANGELOG.rst:
```diff
@@ -11,6 +11,28 @@
+  install`` command.
+- Implement support for building wheels targerting the Python limited
+  API. Extension modules targeting the Python limited API can be
+  easily build starting with the upcomin Meson 1.3.0 release.
```
- build -> built
- upcomin -> upcoming

rgommers commented on CHANGELOG.rst:
```diff
@@ -11,6 +11,28 @@
+- Implement support for building wheels targerting the Python limited
+  API. Extension modules targeting the Python limited API can be
+  easily build starting with the upcomin Meson 1.3.0 release.
+- Instead than silently using the version declared in ``meson.build``,
```
- instead than -> instead of
- perhaps turn it around: _"When `pyproject.toml` does not contain a `version` field and ..."_. Might be a little easier to understand for the reader whether this applies to them.

rgommers commented on CHANGELOG.rst:
```diff
@@ -11,6 +11,28 @@
+  raise an error when ``pyproject.toml`` does not contain a
+  ``version`` field and ``version`` is not declared dynamic.
+- Fix the mtime of source files in the sdist tarball.
+- Fix the installation of files via ``install_subdir()``. For the
```
I suggest extending the first sentence with _"when the `exclude_directories or `exclude_files keywords are used"_. The way it reads now, a user may think there was a broader problem.

dnicolodi replied:
I thought that there were broader problems with handling `install_subdir()` but looking at the old code again, I cannot identify what these problems may have been. I'll update this entry.

rgommers replied:
Nothing important I think - both Pandas and NumPy use it fairly extensively, and it didn't give them problems at least AFAIK.

rgommers commented on CHANGELOG.rst:
```diff
@@ -11,6 +11,28 @@
+  Meson version 1.1.0 or later is required.
+- Add ``objc`` and ``objcpp`` compilers to the cross file generated
+  when the ``$ARCHFLAGS`` is set.
+- Extensive documentation improvements.
```
How about adding here:

- `meson install` is no longer run as part of the wheel build process, resulting in less log being written to stdout.

dnicolodi replied:
I'm not sure it is worth mentioning it, we just got rid of a single line looking something like `+ meson install --quiet --no-rebuild --allow-dirty`

rgommers replied:
maybe not indeed. okay, resolving.

rgommers reviewed (commented):
Thanks @dnicolodi! Looks good overall - a few comments and suggestions.

dnicolodi commented:
I've implemented all suggestions. I also added an entry bout `meson install` after all, and reordered the entries to be grouped more logically, instead of being sorted by implementation date, which is irrelevant to the reader.

dnicolodi commented:
Good idea, but we describe a way to pass arguments to `meson install` also here https://meson-python.readthedocs.io/en/latest/reference/config-settings.html#cmdoption-arg-install-args

rgommers commented:
Ah forgot we already have an issue: gh-473. I'll comment there.

dnicolodi commented:
I've beaten you to it :)
